analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -126,21 +192,530 @@ class Yarn2(
+                                null,
+                                projectAsDependencyPkg.collectDependencies(allScopedDependencies)
+                            )
+                        }
+                    } else {
+                        val packageDependency = allPackages[dependency]
+                        if (packageDependency == null) {
+                            log.warn { "Could not find package for dependency $dependency." }
+                            null
+                        } else {
+                            // As small hack here : Because the detection of dependencies per scope is limited (due to
+                            // the fact it relies on package.json parsing and only the project ones are available), the
+                            // dependencies of a package are always searched in the 'Dependencies' scope, instead of
+                            // the scope of this package.
+                            YarnModuleInfo(
+                                packageDependency.id,
+                                packageDependency,
+                                packageDependency.collectDependencies(allDependencies[DependencyType.DEPENDENCIES]!!)
+                            )
+                        }
+                    }
+                }?.toSet().orEmpty()
+                dependenciesInfo.forEach {
+                    graphBuilder.addDependency(qualifiedScopeName, it)
+                }
+            }
+        }
+        return allProjects
+    }
+
+    /**
+     * Construct a Package or a Project by parsing it [json] representation outputted by `yarn info`, run for the given
+     * [definitionFile]. The objects constructed by this function are put either in [allPackages] or in [allProjects].
+     * Additional data necessary for constructing the instances is fetched in [packagesHeaders] and [packagesDetails].
+     * The list of dependencies of the constructed object is returned.
+     */
+    private fun parsePackage(
+        json: ObjectNode,
+        definitionFile: File
+    ): Map<DependencyType, Pair<Identifier, List<Identifier>>> {
+        val value = json["value"].textValue()
+
+        val header = requireNotNull(packagesHeaders[value]) {
+            "No package header found for '$value'."
+        }
+
+        val (namespace, name) = Npm.splitNamespaceAndName(header.rawName)
+        val childrenNode = json["children"]
+        val version = childrenNode["Version"].textValue()
+
+        val manifest = childrenNode["Manifest"] as ObjectNode
+        // Yarn's manifests contain licenses in an element with an uppercase 'L'. To leverage existing license parsing
+        // code, an extra property with lowercase 'L' is added.
+        manifest["License"].takeUnless {
+            it is NullNode
+        }?.also {
+            manifest.set<ObjectNode>("license", it)
+        }
+        val declaredLicenses = Npm.parseLicenses(manifest)
```

Comment:
Blank line before this one, but in exchange remove them between the single-line properties below.